src/Aspire.Hosting/Dashboard/DashboardServiceHost.cs

```diff
@@ -1,113 +1,188 @@
+using Aspire.Hosting.Publishing;
+using Aspire.Hosting.Utils;
 using Microsoft.AspNetCore.Builder;
 using Microsoft.AspNetCore.Hosting;
+using Microsoft.AspNetCore.Hosting.Server;
+using Microsoft.AspNetCore.Hosting.Server.Features;
+using Microsoft.AspNetCore.Server.Kestrel.Core;
 using Microsoft.Extensions.DependencyInjection;
 using Microsoft.Extensions.Hosting;
-using Microsoft.AspNetCore.Server.Kestrel.Core;
 using Microsoft.Extensions.Logging;
 using Microsoft.Extensions.Options;
-using Aspire.Hosting.Publishing;
-using Aspire.Hosting.Utils;
 
 namespace Aspire.Hosting.Dashboard;
 
 /// <summary>
-/// Hosts a gRPC service (via <see cref="DashboardService"/>) that a dashboard can connect to.
+/// Hosts a gRPC service via <see cref="DashboardService"/> (aka the "Resource Service") that a dashboard can connect to.
 /// Configures DI and networking options for the service.
 /// </summary>
 internal sealed class DashboardServiceHost : IHostedService
 {
+    /// <summary>
+    /// Name of the environment variable that optionally specifies the resource service URL,
+    /// which the dashboard will connect to over gRPC.
+    /// </summary>
+    /// <remarks>
+    /// This is primarily intended for cases outside of the local developer environment.
+    /// If no value exists for this variable, a port is assigned dynamically.
+    /// </remarks>
     private const string DashboardServiceUrlVariableName = "DOTNET_DASHBOARD_GRPC_ENDPOINT_URL";
-    private const string DashboardServiceUrlDefaultValue = "http://localhost:18999";
+
+    /// <summary>
+    /// Provides access to the URI at which the resource service endpoint is hosted.
+    /// </summary>
+    private readonly TaskCompletionSource<string> _resourceServiceUri = new();
 
     /// <summary>
     /// <see langword="null"/> if <see cref="DistributedApplicationOptions.DashboardEnabled"/> is <see langword="false"/>.
     /// </summary>
     private readonly WebApplication? _app;
+    private readonly ILogger<DashboardServiceHost> _logger;
 
     public DashboardServiceHost(
         DistributedApplicationOptions options,
         DistributedApplicationModel applicationModel,
         KubernetesService kubernetesService,
         IOptions<PublishingOptions> publishingOptions,
         ILoggerFactory loggerFactory,
+        ILogger<DashboardServiceHost> logger,
         IConfigureOptions<LoggerFilterOptions> loggerOptions)
     {
-        if (!options.DashboardEnabled)
-        {
-            return;
-        }
-        else if (publishingOptions.Value.Publisher == "manifest")
+        _logger = logger;
+
+        if (!options.DashboardEnabled ||
+            publishingOptions.Value.Publisher == "manifest") // HACK: Manifest publisher check is temporary until DcpHostService is integrated with DcpPublisher.
         {
-            // HACK: Manifest publisher check is temporary until DcpHostService is integrated with DcpPublisher.
+            _resourceServiceUri.SetCanceled();
```

Comment:
Write log that the dashboard service isn't starting?

Reply:
I wanted to see what would happen if the dashboard is disabled but DistributedApplicationOptions's DashboardEnabled/DisableDashboard (it's weird there is two) properties are never set.

Is it for a future feature?

Reply:
Added a log message.

Will kick off a conversation about disabling. Supposedly there's a mechanism for this, though unless there's some reflection at play, the only place that sets `DisableDashboard` (from which `DashboardEnabled` is derived) is in test code.